### PR TITLE
fix: dont terminate the executor for the user

### PR DESCRIPTION
**Pull request by cryptobench**

### CODE REVIEW

File: examples/docs-examples/examples/switching-to-mainnet/run-on-polygon.mjs [hunk 1]
```diff
@@ -7,8 +7,13 @@ import { TaskExecutor } from "@golem-sdk/golem-js";
...
-  const result = await executor.run(async (ctx) => (await ctx.run("node -v")).stdout);
-  await executor.end();
-
-  console.log("Task result:", result);
+  try {
+    const result = await executor.run(async (ctx) => (await ctx.run("node -v")).stdout);
+    console.log("Task result:", result);
+  } catch (err) {
+    console.error("An error occurred:", err);
+    console.error(err);
```
Comment (grisha87) [inline examples/docs-examples/examples/switching-to-mainnet/run-on-polygon.mjs:15]: I don't think we need to log it twice :)

File: examples/docs-examples/examples/transferring-data/upload-file.mjs [hunk 2]
```diff
@@ -11,18 +11,22 @@ import * as fs from "fs";
...
-  console.log("md5 of the file sent to provider: ", result);
-  console.log("Locally computed  md5: ", hash);
+      return res.stdout;
+    });
 
-  await executor.end();
+    console.log("md5 of the file sent to provider: ", result);
+    console.log("Locally computed md5: ", hash);
+  } catch (error) {
+    console.error("Computation failed:", error);
```
Comment (grisha87) [inline examples/docs-examples/examples/transferring-data/upload-file.mjs:28]: I like that you provide context in this log message - that's a good practice. Could you please include it in other examples you modified in this PR?

File: examples/docs-examples/examples/working-with-results/multi-command-endstream.mjs [hunk 3]
```diff
@@ -6,17 +6,23 @@ import { TaskExecutor } from "@golem-sdk/golem-js";
...
+        .downloadFile("/golem/input/output.txt", "./output.txt")
+        .endStream();
 
-    res.on("data", (result) => console.log(result));
-    res.on("error", (error) => console.error(error));
-    res.on("close", () => executor.end());
-  });
+      res.on("data", (result) => console.log(result));
+      res.on("error", (error) => console.error(error));
+      res.on("close", () => {});
```
Comment (grisha87) [inline examples/docs-examples/examples/working-with-results/multi-command-endstream.mjs:21]: I wouldn't change that. The original code was waiting here for the stream to be "closed".

Alternative would be to something like `return new Promise((resolve) => res.once("close", resolve))` instead and then the whole `executor.run` would really wait for this particular event.

File: examples/docs-examples/quickstarts/web-quickstart/requestor.mjs [hunk 4]
```diff
@@ -29,13 +29,15 @@ async function run() {
...
+  });
 
-  await executor
-    .run(async (ctx) => appendResults((await ctx.run("echo 'Hello World'")).stdout))
-    .catch((e) => logger.error(e));
-
-  await executor.end();
+  try {
+    await executor.run(async (ctx) => appendResults((await ctx.run("echo 'Hello World'")).stdout));
+  } catch (e) {
```
Comment (grisha87) [inline examples/docs-examples/quickstarts/web-quickstart/requestor.mjs:36]: Would be good to stay consistent and `catch (err)`.

File: examples/strategy/blackListProvidersIds.ts [hunk 5]
```diff
@@ -16,6 +16,13 @@ const blackListProvidersIds = [
...
     proposalFilter: ProposalFilters.blackListProposalIdsFilter(blackListProvidersIds),
   });
-  await executor.run(async (ctx) => console.log((await ctx.run("echo 'Hello World'")).stdout));
-  await executor.end();
+
+  try {
+    await executor.run(async (ctx) => console.log((await ctx.run("echo 'Hello World'")).stdout));
+  } catch (err) {
+    console.error("Task execution failed:", err);
+    console.error(err);
```
Comment (grisha87) [inline examples/strategy/blackListProvidersIds.ts:24]: Double log strikes once again.

**Review by grisha87 — CHANGES_REQUESTED** [review]

File: examples/docs-examples/examples/composing-tasks/batch-endstream-chunks.mjs [hunk 6]
```diff
@@ -6,17 +6,21 @@ import { TaskExecutor } from "@golem-sdk/golem-js";
...
+        .beginBatch()
+        .uploadFile("./worker.mjs", "/golem/input/worker.mjs")
+        .run("node /golem/input/worker.mjs > /golem/input/output.txt")
+        .run("cat /golem/input/output.txt")
+        .downloadFile("/golem/input/output.txt", "./output.txt")
+        .endStream();
 
-    res.on("data", (data) => (data.index == 2 ? console.log(data.stdout) : ""));
-    res.on("error", (error) => console.error(error));
-    res.on("close", () => executor.end());
```
Comment (grisha87) [inline examples/docs-examples/examples/composing-tasks/batch-endstream-chunks.mjs:20]: This sill requires fixing. We want to wait for the stream to close, before we `end/shutdown` the executor.

**Review by grisha87 — CHANGES_REQUESTED** [review]

File: examples/docs-examples/examples/composing-tasks/batch-endstream-chunks.mjs (hunk 6 above)
Comment (cryptobench) [reply examples/docs-examples/examples/composing-tasks/batch-endstream-chunks.mjs:20]: Please check the updated code. Is that what you meant?

Comment (grisha87) [reply examples/docs-examples/examples/composing-tasks/batch-endstream-chunks.mjs:20]: Not really :) You did it already in the other place by:

```ts
res.once("close", resolve);
```

Comment (cryptobench) [reply examples/docs-examples/examples/composing-tasks/batch-endstream-chunks.mjs:20]: @grisha87 I must admit i'm not sure what to do from here. I am not sure I understand what you want me to do.

**Comment by cryptobench** [comment]
its ready @grisha87

**Review by grisha87 — APPROVED** [review]

**Pull request merged**